src/sctools/test/test_metrics.py

```diff
@@ -136,24 +138,33 @@ def test_spliced_reads(gene_metrics):
     assert observed == expected
 
 
-def test_higher_order_metrics(gene_metrics):
+def test_higher_order_metrics_mean(gene_metrics):
+    print(gene_metrics["reads_per_fragment"])
     higher_order_metrics = [
         ('molecule_barcode_fraction_bases_above_30_mean', float),
         ('molecule_barcode_fraction_bases_above_30_variance', float),
-        ('genomic_reads_fraction_bases_quality_above_30_mean', float),
-        ('genomic_reads_fraction_bases_quality_above_30_variance', float),
-        ('genomic_read_quality_mean', float),
-        ('genomic_read_quality_variance', float),
-        ('n_molecules', int),
-        ('n_fragments', int),
+        # ('genomic_reads_fraction_bases_quality_above_30_mean', float),  # These four tests are commented out because
+        # ('genomic_reads_fraction_bases_quality_above_30_variance', float), # the notebook equivalent dont ignore soft clipped bases
+        # ('genomic_read_quality_mean', float),
+        # ('genomic_read_quality_variance', float)
         ('reads_per_molecule', float),
         ('reads_per_fragment', float),
-        ('fragments_per_molecule', float),
-        ('fragments_with_single_read_evidence', int),
-        ('molecules_with_single_read_evidence', int),
+        ('fragments_per_molecule', float)
     ]
     for (key, func) in higher_order_metrics:
         observed = gene_metrics[key].mean()
         expected = func(_testing_knowledge[key])
-        assert observed == expected, '%s failed' % key
+        assert math.isclose(observed, expected), '%s failed' % key
 
+
+def test_higher_order_metrics_sum(gene_metrics):
```

Comment:
Lets test equivalence for each of these and output a longer error message that includes all the failing tests at once. 

Something like: 
```
test_results = {}
for (key, func) in higher_order_metrics:
    observed = gene_metrics[key].sum()
    expected = func(_testing_knowledge[key])
    test_results[key] = (observed == expected, observed, expected)
# lazy but fine: print dictionary of results if any test fails. 
assert all(v[0] for v in test_results.values()), repr(test_results)  
```